File: Baekjoon/Implementation/[13414]수강신청/solution.js
const fileName = process.platform === "linux" ? "/dev/stdin" : "example.dat";
const inputData = require("fs")
  .readFileSync(fileName)
  .toString()
  .trim()
  .split("\n");

const input = (() => {
  let line = 0;
  return () => inputData[line++];
})();

let [K, L] = input()
  .split(" ")
  .map((item) => Number(item));
const enroll = {};

for (let i = 0; i < L; i++) {
  enroll[input()] = i;
}

const res = Object.entries(enroll)
  .sort((a, b) => a[1] - b[1])
  .splice(0, K)
  .map((item) => item[0]);
console.log(res.join("\n"));

File: Baekjoon/Implementation/[13414]수강신청/solution.py
import sys

K, L = map(int, sys.stdin.readline().split())
enroll = dict()
res = []

for i in range(L):
    sid = sys.stdin.readline()

    enroll[sid] = i

for key, value in enroll.items():
    res.append([key, value])

res.sort(key=lambda x: (x[1]))

for i in range(len(res)):
    if K > 0:
        sys.stdout.write(res[i][0])
    else:
        break

    K -= 1
